{
	"documenttitle": {
		"text": "Red Hat Security Advisory: Red Hat support for Spring Boot 2.3.10 security update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2021:3425"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2021-09-09T06:18:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2021-09-09T06:18:00Z",
		"CurrentReleaseDate": "2021-09-09T06:18:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.0.2",
			"Date": "2021-09-09T11:18:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nAn update is now available for Red Hat support for Spring Boot.\n\nRed Hat Product Security has rated this update as having a security impact of Important. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability. For more information, see the CVE links in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nRed Hat support for Spring Boot provides an application platform that reduces the complexity of developing and operating applications (monoliths and microservices) for OpenShift as a containerized platform.\n\nThis release of Red Hat support for Spring Boot 2.3.10 serves as a replacement for Red Hat support for Spring Boot 2.3.6, and includes security and bug fixes and enhancements. For more information, see the release notes listed in the References section.\n\nSecurity Fix(es):\n\n* undertow: special character in query results in server errors (CVE-2020-27782)\n\n* undertow: buffer leak on incoming websocket PONG message may lead to DoS (CVE-2021-3690)\n\n* tomcat: Information disclosure when using NTFS file system (CVE-2021-24122)\n\n* tomcat: Request mix-up with h2c (CVE-2021-25122)\n\n* tomcat: Incomplete fix for CVE-2020-9484 (RCE via session persistence) (CVE-2021-25329)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2021 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Important",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2021:3425",
				"Description": "https://access.redhat.com/errata/RHSA-2021:3425"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#important",
				"Description": "https://access.redhat.com/security/updates/classification/#important"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=catRhoar.spring.boot&version=2.3.10",
				"Description": "https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=catRhoar.spring.boot&version=2.3.10"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/documentation/en-us/red_hat_support_for_spring_boot/2.3/html/release_notes_for_spring_boot_2.3/index",
				"Description": "https://access.redhat.com/documentation/en-us/red_hat_support_for_spring_boot/2.3/html/release_notes_for_spring_boot_2.3/index"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "undertow: special character in query results in server errors",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-01-25T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2020-27782",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3425"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2020-27782",
						"Description": "CVE-2020-27782"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1901304",
						"Description": "bz#1901304: CVE-2020-27782 undertow: special character in query results in server errors"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2020-11-24T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "undertow: buffer leak on incoming websocket PONG message may lead to DoS",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-07-30T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3690",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3425"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3690",
						"Description": "CVE-2021-3690"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1991299",
						"Description": "bz#1991299: CVE-2021-3690 undertow: buffer leak on incoming websocket PONG message may lead to DoS"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-08-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "tomcat: Information disclosure when using NTFS file system",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-01-14T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-24122",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3425"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "http://mail-archives.apache.org/mod_mbox/tomcat-announce/202101.mbox/%3Cf3765f21-969d-7f21-e34a-efc106175373%40apache.org%3E",
						"Description": "http://mail-archives.apache.org/mod_mbox/tomcat-announce/202101.mbox/%3Cf3765f21-969d-7f21-e34a-efc106175373%40apache.org%3E"
					},
					{
						"URL": "http://tomcat.apache.org/security-7.html#Fixed_in_Apache_Tomcat_7.0.107",
						"Description": "http://tomcat.apache.org/security-7.html#Fixed_in_Apache_Tomcat_7.0.107"
					},
					{
						"URL": "http://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.60",
						"Description": "http://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.60"
					},
					{
						"URL": "http://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.40",
						"Description": "http://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.40"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-24122",
						"Description": "CVE-2021-24122"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1917209",
						"Description": "bz#1917209: CVE-2021-24122 tomcat: Information disclosure when using NTFS file system"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-01-15T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "4",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "tomcat: Request mix-up with h2c",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-03-01T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-25122",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3425"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "http://mail-archives.apache.org/mod_mbox/tomcat-announce/202103.mbox/%3Cb7626398-5e6d-1639-4e9e-e41b34af84de%40apache.org%3E",
						"Description": "http://mail-archives.apache.org/mod_mbox/tomcat-announce/202103.mbox/%3Cb7626398-5e6d-1639-4e9e-e41b34af84de%40apache.org%3E"
					},
					{
						"URL": "https://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.2",
						"Description": "https://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.2"
					},
					{
						"URL": "https://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.63",
						"Description": "https://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.63"
					},
					{
						"URL": "https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.43",
						"Description": "https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.43"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-25122",
						"Description": "CVE-2021-25122"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1934032",
						"Description": "bz#1934032: CVE-2021-25122 tomcat: Request mix-up with h2c"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-03-02T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "5",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "tomcat: Incomplete fix for CVE-2020-9484 (RCE via session persistence)",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-03-01T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-25329",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3425"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "http://mail-archives.apache.org/mod_mbox/tomcat-announce/202103.mbox/%3C811bba77-e74e-9f9b-62ca-5253a09ba84f%40apache.org%3E",
						"Description": "http://mail-archives.apache.org/mod_mbox/tomcat-announce/202103.mbox/%3C811bba77-e74e-9f9b-62ca-5253a09ba84f%40apache.org%3E"
					},
					{
						"URL": "https://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.2",
						"Description": "https://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.2"
					},
					{
						"URL": "https://tomcat.apache.org/security-7.html#Fixed_in_Apache_Tomcat_7.0.108",
						"Description": "https://tomcat.apache.org/security-7.html#Fixed_in_Apache_Tomcat_7.0.108"
					},
					{
						"URL": "https://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.63",
						"Description": "https://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.63"
					},
					{
						"URL": "https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.43",
						"Description": "https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.43"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-25329",
						"Description": "CVE-2021-25329"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1934061",
						"Description": "bz#1934061: CVE-2021-25329 tomcat: Incomplete fix for CVE-2020-9484 (RCE via session persistence)"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2021-03-02T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {}
}
